### PR TITLE
profiler handlers added

### PR DESCRIPTION
**Pull request by sekulicd**
This enables profiler that can be used for inspecting bottlenecks in the application.
It can be disabled via configuration and it shouldn't be used in production.

Here are the commands that can be used to view different profiles:
go tool pprof http://localhost:8024/debug/pprof/profile\?seconds\=30
go tool pprof http://localhost:8024/debug/pprof/heap
go tool pprof http://localhost:8024/debug/pprof/block
wget http://localhost:8024/debug/pprof/trace?seconds=5
go tool pprof http://localhost:8024/debug/pprof/mutex
Overview:
http://localhost:8024/debug/pprof/

@tiero @altafan please review.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -46,6 +46,10 @@ const (
 	SSLKeyPathKey = "SSL_KEY"
 	// MnemonicKey is the mnemonic of the master private key of the daemon's wallet
 	MnemonicKey = "MNEMONIC"
+	// Enables profiler that can be used to investigate performance issues
+	EnableProfiler = "ENABLE_PROFILER"
+	// ProfilerPort defines profiler port
+	ProfilerPort = "PROFILER_PORT"
```
Comment (tiero) [inline config/config.go:52]: I think for sake of simplicity we just decided an hardcoded port and be with it 

```suggestion
```

File: config/config.go [hunk 2]
```diff
@@ -46,6 +46,10 @@ const (
 	SSLKeyPathKey = "SSL_KEY"
 	// MnemonicKey is the mnemonic of the master private key of the daemon's wallet
 	MnemonicKey = "MNEMONIC"
+	// Enables profiler that can be used to investigate performance issues
+	EnableProfiler = "ENABLE_PROFILER"
```
Comment (tiero) [inline config/config.go:50]:
```suggestion
	EnableProfilerKey = "ENABLE_PROFILER"
```

File: config/config.go [hunk 3]
```diff
@@ -68,6 +72,8 @@ func init() {
 	vip.SetDefault(TradeExpiryTimeKey, 120)
 	vip.SetDefault(DataDirPathKey, defaultDataDir)
 	vip.SetDefault(PriceSlippageKey, 0.05)
+	vip.SetDefault(EnableProfiler, false)
+	vip.SetDefault(ProfilerPort, 8024)
```
Comment (tiero) [inline config/config.go:76]:
```suggestion
	vip.SetDefault(EnableProfilerKey, false)
```

File: cmd/tdexd/main.go [hunk 4]
```diff
@@ -31,11 +32,24 @@ import (
...
 
 func main() {
 	log.SetLevel(log.Level(config.GetInt(config.LogLevelKey)))
 
+	//http://localhost:{config.ProfilerPort}}/debug/pprof/
+	if config.GetBool(config.EnableProfiler) {
+		runtime.SetBlockProfileRate(1)
+		go func() {
+			http.ListenAndServe(
+				fmt.Sprintf(":%v", config.GetString(config.ProfilerPort)),
```
Comment (tiero) [inline cmd/tdexd/main.go:47]: hardcode here

File: config/config.go (hunk 1 above)
Comment (sekulicd) [reply config/config.go:52]: Done

File: config/config.go (hunk 2 above)
Comment (sekulicd) [reply config/config.go:50]: Done

File: config/config.go (hunk 3 above)
Comment (sekulicd) [reply config/config.go:76]: Done

File: cmd/tdexd/main.go (hunk 4 above)
Comment (sekulicd) [reply cmd/tdexd/main.go:47]: Done

**Comment by tiero** [comment]
Solve conflicts @sekulicd

**Pull request merged**